### ISSUE TITLE
Failed to load script, compilation failed

### ISSUE THREAD

**Issue by MrCythos** [poohcom1/godot-animated-sprite-2-player#10]
I'm getting an error when I try to load this plugin saying that the compilation is failing. This is the following output in the debugger.

  res://addons/AS2P/InspectorConvertor.gd:1 - Parse Error: Unexpected "Identifier" in class body.
  res://addons/AS2P/plugin.gd:-1 - Compile Error: 
  modules/gdscript/gdscript.cpp:2788 - Failed to load script "res://addons/AS2P/plugin.gd" with error "Compilation failed". (User)
  res://addons/AS2P/plugin.gd:9 - Invalid call. Nonexistent function 'new' in base 'GDScript'.

I'm on v4.2.1.stable.official [b09f793f5]

**Comment by poohcom1** [comment]
I can't seem to reproduce the compiler error. Can you try opening up `res://addons/AS2P/InspectorConvertor.gd` in the script editor and send me a screenshot of the highlighted error?

**Comment by MrCythos** [comment]
After I opened it, it looks like for whatever reason, when I downloaded it off of godot's site, the scripts were for godot 3 despite selecting the 4.0 version. After you asked me to open up those files, the errors that popped up were things like "tool" instead of "@tool" and "yield" instead of "await". I was able to fix it by copy-pasting the code directly from the github page and replacing everything for all three .gd scripts. 

![image](https://github.com/poohcom1/godot-animated-sprite-2-player/assets/5875251/9f9514f4-88ca-4068-95cd-9257293e3602)
